### PR TITLE
fix: pinned reply to deleted msg fix #14510

### LINKED ISSUE #14510: The message is not shown as deleted in the pinned section if 'delete for everyone' option is used
**Steps to reproduce:**
1. Go to the 1-1 chat
2. Send the message: "Test_A"
3. Leave a reply: "Test_B"
4. Pin the message
5. Remove replied message "Test_A"
6. Open the list of pinned messages
7. Remove the pinned message using 'delete for everyone' option
8. Check the replied message

**Actual result:**
The replied message "Test_A" is shown after removing

https://user-images.githubusercontent.com/52490791/214904661-2b3cea83-a4c0-404d-a67d-8acd01d516b9.mp4

image

**Expected result:**
The pinned message is shown as removed in the chat
image

### PR DESCRIPTION
**Pull request by yqrashawn**
resolve #14510

show "Message deleted" pinned replied message

image

status: ready

### CODE REVIEW

File: src/status_im2/contexts/chat/messages/pin/list/view.cljs [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 (defn message-render-fn
-  [{:keys [whisper-timestamp] :as message}
+  [{:keys [whisper-timestamp chat-id] :as message}
    _
    {:keys [group-chat public? community? current-public-key show-input? edit-enabled]}]
   ;; TODO (flexsurfer) probably we don't want reactions here
   [message/message-with-reactions
    message
    {:group-chat          group-chat
+    :chat-id             chat-id
```
Comment (yqrashawn) [inline src/status_im2/contexts/chat/messages/pin/list/view.cljs:20]: required in https://github.com/status-im/status-mobile/blob/d375758d235263e73040929efdbf597a7ac2049e/src/status_im2/contexts/chat/messages/content/view.cljs#L98

**Comment by Parveshdhull** [comment]
Hi @yqrashawn, After sender deletes the original message, if replier clicks on pinned reply message, it is giving this error



Please can you take a look

**Comment by Parveshdhull** [comment]
> Hi @yqrashawn, After deleting the original message and clicking on pinned reply message, it is giving this error

Issue also exists in develop

**Comment by yqrashawn** [comment]
> > Hi @yqrashawn, After deleting the original message and clicking on pinned reply message, it is giving this error
> 
> Issue also exists in develop

There's indeed reproducible crash in dev. The error is can't get contact-name[0], caused by this line https://github.com/status-im/status-mobile/blob/d198c4bde0e43d5caa97788709b386eefb9d108a/src/status_im/ui2/screens/chat/components/reply/view.cljs#L36
This PR fixed the crash as well as the display issue.

I just tested on both iOS and android and can't reproduce the issue after this PR.

**Review by Parveshdhull — APPROVED** [review]

**Comment by pavloburykh** [comment]
@yqrashawn thanx for the PR. Could you please rebase PR to latest develop. There is a commit https://github.com/status-im/status-mobile/commit/e4722e05f3289d50349ebe4cf0b420e14b24f578 which would be better to include before testing. Thanx!

**Comment by yqrashawn** [comment]
> @yqrashawn thanx for the PR. Could you please rebase PR to latest develop. There is a commit [e4722e0](https://github.com/status-im/status-mobile/commit/e4722e05f3289d50349ebe4cf0b420e14b24f578) which would be better to include before testing. Thanx!

Rebased to dev

**Comment by pavloburykh** [comment]
> This PR fixed the crash as well as the display issue.
> I just tested on both iOS and android and can't reproduce the issue after this PR.

@yqrashawn Unfortunately crash still exists. 

I haven't figured out preconditions for this crash yet, but I have user with community in which crash is reproducible every time. As a result of crash - it is impossible to enter community channel with any user even after app relaunch, re-install etc. So it is a critical issue. 

Reproducible both for IOS and Android. 

Here is **seed phrase** of user with crashing community: document cabbage search chalk rich gorilla elephant cricket prize require erase neutral

**Steps:**

1. Install Status PR build
2. Recover multiaccount via given seedphrase
3. Tap Communities tab
4. Find in the list community with the name "hjkkk"
5. Open community
6. Open "general" channel
7. Wait for few seconds
8. Observe crash

**Actual result:** app crashes

**Expected result:** no crash

https://user-images.githubusercontent.com/97245802/216640976-1b4b5a87-529d-41e6-94db-6fce4178dcd0.mp4

[crash_logs.zip](https://github.com/status-im/status-mobile/files/10580243/crash_logs.zip)

@yqrashawn let me know if above details are enough for figuring out the rootcause. If not - I will continue investigate steps.

**Comment by yqrashawn** [comment]
@Parveshdhull @pavloburykh This is fixed in https://github.com/status-im/status-mobile/pull/14977

**Pull request closed without merging**